### PR TITLE
779: Dockerize api-mobile

### PR DESCRIPTION
**Pull request by NickPhura**
- Add docker files for api-mobile
- Add root level docker files to run all applications using shared db (api, app, api-mobile)

### From `./api-mobile`
- Can run `make local` to kick-off a number of docker containers that will run:
  - postgres
  - the new api-mobile, which will run its migrations on startup
_Note: this doesn't include any of the seeding or migrations from the original `./api`_

Can access api-mobile at: `localhost/api`

See `./api-mobile/Makefile` for all the commands (essentially the same as the original `./api/Makefile`

### From `./` (root)
- Can run `make local` to kick off a number of docker containers that will run:
  - postgres
  - the original api, which will run its migrations and seeding on startup
  - the web-app
  - the new api-mobile, which will run its migrations on startup

Can access api at: `localhost/api`
Can access api-mobile at: `localhost/api-mobile/`
Can access web-app at: `localhost:3033`

See `./Makefile` for all the commands (essentially the same as the original `./api/Makefile`

### CODE REVIEW

**Review by rstens — COMMENTED** [review]
Just a comment on versions, the solution currently runs node v10.14.1 not 11

**Review by rstens — COMMENTED** [review]
For the database, you could simply use: https://github.com/bcgov/openshift-postgresql-oracle_fdw/tree/master/centos7.rh-postgresql96

it is also what our prod db is based on (The rh image that is)

File: api-mobile/README.md [hunk 1]
```diff
@@ -1,14 +1,48 @@
 # bcgov/lucy-web/api-mobile
 
-# Technologies Used
+## Technologies Used
 
 | Technology | Version | Website                              | Description          |
 | ---------- | ------- | ------------------------------------ | -------------------- |
 | node       | 10.x.x  | https://nodejs.org/en/               | JavaScript Runtime   |
 | npm        | 6.x.x   | https://www.npmjs.com/               | Node Package Manager |
 | PostgreSQL | 10.x    | https://www.postgresql.org/download/ | PSQL database        |
```
Comment (rstens) [inline api-mobile/README.md]: We are actually using psql 9.6

File: api-mobile/README.md [hunk 2]
```diff
@@ -32,7 +137,9 @@ Linting and formatting is handled by a combiation of `ESlint` and `Prettier`. Th
 
 ### Technolgies used
```
Comment (rstens) [inline api-mobile/README.md]: Typo in Technolgies

File: api-mobile/app.ts [hunk 3]
```diff
@@ -11,7 +11,8 @@ const defaultLog = getLogger('app');
 
 import { authenticate } from './src/utils/auth-utils';
 
-const HOST = process.env.API_HOST || 'localhost:3002';
+const HOST = process.env.API_HOST || 'localhost';
+const PORT = Number(process.env.API_PORT || '3002');
```
Comment (rstens) [inline api-mobile/app.ts:15]: Why aren't we simply using 3001? We are in a different POD.

**Review by rstens — COMMENTED** [review]
I am missing the Migrations update that caused the deploy to Dev to fail.

**Comment by NickPhura** [comment]
> 
> 
> I am missing the Migrations update that caused the deploy to Dev to fail.

After looking into the migrations issue more, the reason they are failing is because in Dev, we had previously manually created the tables that are now handled by the migrations.  And so the migrations scripts try to run (because the `migration` table indicates they have never run before) and then fail because the tables are already there when they shouldn't be.  I just need to go in and drop those tables, and let the migrations re-create them.

Edit: chatted on teams, Roland removed the culprit table from Dev.

File: api-mobile/app.ts (hunk 3 above)
Comment (NickPhura) [reply api-mobile/app.ts:15]: Mainly so that if you run it locally, it doesn't clash with the other api

**Comment by NickPhura** [comment]
> 
> 
> Just a comment on versions, the solution currently runs node v10.14.1 not 11



> 
> 
> For the database, you could simply use: https://github.com/bcgov/openshift-postgresql-oracle_fdw/tree/master/centos7.rh-postgresql96
> 
> it is also what our prod db is based on (The rh image that is)

In regards to the `./api/.docker/db/Dockerfile`, I actually am not sure when that gets used.  I don't think it runs as part of the local docker setup, since `./api/.docker/db/Dockerfile.local` is what gets referenced in the docker-compose file.

I am more than happy to update the contents of that Dockerfile to the one you linked though, if that is a better version? I don't know much about the options there, for postgres container setups, so whatever you think is best sounds good to me.

Let me know what your thoughts are though.

Edit: chatted on teams, no issue here at the moment.

**Comment by NickPhura** [comment]
> 
> 
> Just a comment on versions, the solution currently runs node v10.14.1 not 11

I've updated the node version used in several of the dockerfiles from Node 11 to 10, to be consistent.

**Review by micheal-w-wells — APPROVED** [review]

**Pull request merged**